### DISCUSSION TITLE
Loss of METviewer Database

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#1702] [Plot Generation] [answered]
Hi all,

I had a peculiar incident with METviewer that I could use some help with. I am running METviewer v4.1.0 on an AWS instance. I have a database that has data on it. Yesterday morning, I created plots with data from that database. In the afternoon, the data was gone. I ran the plots in the morning using the 'mv_batch' command. In the afternoon, I did the same, but this time I got the following error:

`No data was returned from database
2022-07-21 18:32:48 edu.ucar.metviewer.db.mysql.MysqlAppDatabaseManager.printFormattedTable(Unknown Source) -1 printFormattedTable -   **  ERROR: Caught class edu.ucar.metviewer.EmptyResultSetException in printFormattedTable(ResultSet res): result set contained no data
  **  ERROR: Caught class edu.ucar.metviewer.EmptyResultSetException in printFormattedTable(ResultSet res): result set contained no data
  **  ERROR: Caught class edu.ucar.metviewer.EmptyResultSetException in printFormattedTable(ResultSet res): result set contained no data`
  
When I went to the GUI, the variables were there, but for metrics, there was one option `NA`. 
  
 Myself and @greenlaw were unable to find anything in the logs to determine why the data disappeared. 
  
 Your help is much appreciated. 
  
  Best,
  Lindsay

**Comment by JohnHalleyGotway** [comment]
@TatianaBurek and/or @venitahagerty, can you please take a look at this discussion? Anything advice for Lindsey on where to look to debug what may have happened to that database?

@lindsayrblank, when faced with database issues in the past, I have usually logged onto a machine with access to that database and run mysql commands directly on the command line. I do that to confirm that the expected database actually exists and contains data.

**Comment by TatianaBurek** [comment, marked as answer]
@lindsayrblank  the error that you posted indicates that the table(s) in the database don't have data that would satisfy your criteria or the table(s)  are empty.
METviewer GUI usually caches previous  query results for filling the menus so the web page might display old database state.
I would follow John's advise and check the database directly using SQL commands. Let me know if you need help with this

**Reply by lindsayrblank** [reply]
Thank you both @JohnHalleyGotway and @TatianaBurek for the responses. What commands would I use to check the database directly?

**Reply by TatianaBurek** [reply]
To connect to the database:
`mysql -u<user_name> -p<pwd> -h<server_name> <database_name>`

To check the  data:
`select * from <table_name>;`
where table_name depends on the data type. For example: line_data_vl1l2, line_data_pct, line_data_cts

**Reply by TatianaBurek** [reply]
@lindsayrblank  do you still have problems with the data? Were you able to connect to the database?

**Reply by lindsayrblank** [reply]
@TatianaBurek Thank you for replying. I apologize for not getting back to you. I loaded the data into a new database when I began this discussion. I had deleted the old database earlier this week before I was able to try the commands you suggested. If this happens again in the future, I will be sure to use what you suggested here (although hopefully it won't happen again!). Thank you both.
